task_processing/plugins/kubernetes/task_config.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from pyrsistent import field
+from pyrsistent import PMap
+from pyrsistent import PVector
+
+from task_processing.interfaces.task_executor import DefaultTaskConfigInterface
+
+
+class KubernetesTaskConfig(DefaultTaskConfigInterface):
+    node_selector = field(type=PMap)
+    containers = field(type=PVector)
+    # Hardcoded for the time being
+    restart_policy = "Never"
+    # By default, the retrying executor retries 3 times. This task option
+    # overrides the executor setting.
+    retries = field(type=int,
+                    factory=int,
+                    mandatory=False,
+                    invariant=lambda r: (r >= 0, 'retries >= 0'))
+    volumes = field(type=PVector)
+
+    @property
+    def pod_name(self) -> str:
+        return self.name
+
+    def set_pod_name(self, pod_name: str):
+        try:
+            name, uuid = pod_name.rsplit('.', maxsplit=1)
+        except ValueError:
+            raise ValueError(f'Invalid format for pod_name {pod_name}')
+
+        if len(pod_name) > 253:
```

Comment:
can we make a named constant for this?
```suggestion
        MAX_POD_NAME_LENGTH = 253
        ...
        if len(pod_name) > MAX_POD_NAME_LENGTH:
```

Reply:
Changed into invariant.